ProcessLib/ThermoHydroMechanics/ThermoHydroMechanicsFEM.h

```diff
@@ -0,0 +1,784 @@
+                 dNdx_T.transpose() * velocity * N_p * rho_fr * C_f) *
+                w;
+            // coeff matrix using for RHS
+            double heat_capacity =
+                porosity * C_f * rho_fr + (1 - porosity) * C_s * rho_sr;
+            MTT.noalias() += N_T.transpose() * heat_capacity * N_T * w;
+
+            //
+            // temperature equation, pressure part !!!!!!.positive or negative
+            //
+            KTp.noalias() += K_over_mu * rho_fr * C_f * dNdx_p.transpose() *
+                             (dNdx_T * T) * N_T * w * 0;
+        }
+        // temperature equation, temperature part
+        local_Jac
+            .template block<temperature_size, temperature_size>(
+                temperature_index, temperature_index)
+            .noalias() += KTT + MTT / dt;
+
+        // temperature equation, pressure part
+        local_Jac
+            .template block<temperature_size, pressure_size>(temperature_index,
+                                                             pressure_index)
+            .noalias() -= KTp;
+        // displacement equation, temperature part
+        local_Jac
+            .template block<displacement_size, temperature_size>(
+                displacement_index, temperature_index)
+            .noalias() -= KuT;
+
+        // displacement equation, pressure part
+        local_Jac
+            .template block<displacement_size, pressure_size>(
+                displacement_index, pressure_index)
+            .noalias() -= Kup;
+
+        // pressure equation, temperature part.
+        local_Jac
+            .template block<pressure_size, temperature_size>(pressure_index,
+                                                             temperature_index)
+            .noalias() -= storage_T / dt;
+
+        // pressure equation, pressure part.
+        local_Jac
+            .template block<pressure_size, pressure_size>(pressure_index,
+                                                          pressure_index)
+            .noalias() += laplace_p + storage_p / dt;
+
+        // pressure equation, displacement part.
+        local_Jac
+            .template block<pressure_size, displacement_size>(
+                pressure_index, displacement_index)
+            .noalias() += Kup.transpose() / dt;
+
+        // pressure equation (f_p)
+        local_rhs.template segment<pressure_size>(pressure_index).noalias() -=
+            laplace_p * p + storage_p * p_dot - storage_T * T_dot +
+            Kup.transpose() * u_dot;
+
+        // displacement equation (f_u)
+        local_rhs.template segment<displacement_size>(displacement_index)
+            .noalias() += Kup * p;
+
+        // temperature equation (f_T)
+        local_rhs.template segment<temperature_size>(temperature_index)
+            .noalias() -= KTT * T + MTT * T_dot;
+    }
+
+    void postTimestepConcrete(std::vector<double> const& local_x) override
+    {
+        double const& t = _process_data.t;
```

Comment:
Move the this velocity calculation to an overloaded member:
```
  void computeSecondaryVariableConcrete(
        double const /*t*/, std::vector<double> const& local_x) override;
```
and overload
```
computeSecondaryVariableConcrete
```
in class ThermoHydroMechanicsProcess to call ThermoHydroMechanicsLocalAssembler::computeSecondaryVariableConcrete.